packages/hub/routes/job-tickets.ts

```diff
@@ -69,6 +73,81 @@ export default class JobTicketsRoute {
     ctx.type = 'application/vnd.api+json';
     ctx.status = 200;
   }
+
+  async retry(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let id = ctx.params.id;
+    let jobTicketToRetry = await this.jobTicketsQueries.find(id);
+
+    if (!jobTicketToRetry) {
+      ctx.body = {
+        errors: [
+          {
+            status: '404',
+            title: 'Job ticket not found',
+            detail: `Could not find the job ticket ${id}`,
+          },
+        ],
+      };
+      ctx.type = 'application/vnd.api+json';
+      ctx.status = 404;
+
+      return;
+    }
+
+    let requestingUserAddress = ctx.state.userAddress;
+
+    if (jobTicketToRetry.ownerAddress !== requestingUserAddress) {
+      ctx.body = {
+        errors: [
+          {
+            status: '401',
+            title: 'No valid auth token',
+          },
+        ],
+      };
+      ctx.type = 'application/vnd.api+json';
+      ctx.status = 401;
+
+      return;
+    }
+
+    if (jobTicketToRetry.state !== 'failed') {
+      ctx.body = {
+        errors: [
+          {
+            status: '422',
+            title: 'Job ticket state is not failed',
```

Comment:
A nitpick - would prefer `Can only retry a job with failed state (current state: ${jobTicketToRetry.state})` for a little more clarity

Reply:
done in 3959414, thanks